crates/flight_client/src/lib.rs

```diff
@@ -118,40 +118,48 @@ impl std::fmt::Display for TonicStatusMessage {
 
 #[derive(Debug, Snafu)]
 pub enum Error {
-    #[snafu(display("Failed to connect to the Flight server. A TLS error occurred.\n{source}"))]
+    #[snafu(display(
+        "Failed to connect to server: TLS error.\n{source}\nEnsure the flight endpoint is valid and reachable."
+    ))]
     UnableToConnectToServer { source: tls::Error },
 
-    #[snafu(display("Failed to execute query. Invalid metadata value.\n{source}"))]
+    #[snafu(display("Authentication failed.\n{source}\nEnsure the credentials are valid."))]
     InvalidMetadata {
         source: tonic::metadata::errors::InvalidMetadataValue,
     },
 
-    #[snafu(display("Failed to connect to the Flight server.\n{source}"))]
+    #[snafu(display(
+        "Failed to connect to Flight server: Handshake failed.\n{source}\nVerify the configuration and try again."
```

Comment:
```suggestion
        "Failed to connect to Flight server: Handshake failed.\n{source}"
```

The new tonic status messages print their own actions, so let's not repeat an action here.